src/node/gov.h

```diff
@@ -5,26 +5,58 @@
 #include "ds/json.h"
 #include "entities.h"
 #include "kv/map.h"
+#include "proposals.h"
 
 namespace ccf
 {
   namespace jsgov
   {
     using ProposalId = std::string;
-    using Proposal = std::string;
+
     struct ProposalInfo
     {
       ccf::MemberId proposer_id;
+      ccf::ProposalState state;
       std::unordered_map<ccf::MemberId, std::string> ballots = {};
     };
-    DECLARE_JSON_TYPE(ProposalInfo)
-    DECLARE_JSON_REQUIRED_FIELDS(ProposalInfo, proposer_id, ballots);
+    DECLARE_JSON_TYPE(ProposalInfo);
+    DECLARE_JSON_REQUIRED_FIELDS(ProposalInfo, proposer_id, state, ballots);
+
+    struct ProposalSubmitted
```

Comment:
This is the type we return on submission of a proposal or a ballot. It's really a proposal state, but of course `ccf::ProposalState` is already the enum. So maybe `ProposalInstanceState`? Or `jsgov::ProposalState`, not to be confused with `ccf::ProposalState`? Input welcome.

Reply:
Why not return `ProposalInfo`?

Reply:
Going with:

```
    struct ProposalInfoSummary
    {
      ProposalId proposal_id;
      ccf::MemberId proposer_id;
      ccf::ProposalState state;
      size_t ballot_count;
    };
```